src/components/sidebar/right-sidebar/new-datepicker/components.tsx

```diff
@@ -0,0 +1,255 @@
+  position: "absolute",
+  width: "100%",
+  height: 40,
+  top: -13,
+  borderRadius: 7,
+  cursor: "pointer",
+  opacity: 0.3,
+  zIndex: 300,
+};
+
+const railCenterStyle: React.CSSProperties = {
+  position: "absolute",
+  width: "100%",
+  height: 5,
+  borderRadius: 7,
+  cursor: "pointer",
+  pointerEvents: "none",
+  backgroundColor: "#eaeaea"
+};
+
+interface TooltipRailProps {
+  activeHandleID: string;
+  getRailProps: (props: object) => object;
+  getEventData: (e: Event) => object;
+}
+
+export class TooltipRail extends Component<TooltipRailProps> {
+  state = {
+    value: null,
+    percent: null
+  };
+
+  static defaultProps = {
+    disabled: false
+  };
+
+  onMouseEnter = () => {
+    document.addEventListener("mousemove", this.onMouseMove);
+  };
+
+  onMouseLeave = () => {
+    this.setState({ value: null, percent: null });
+    document.removeEventListener("mousemove", this.onMouseMove);
+  };
+
+  onMouseMove = (e: Event) => {
+    const { activeHandleID, getEventData } = this.props;
+
+    if (activeHandleID) {
+      this.setState({ value: null, percent: null });
+    } else {
+      this.setState(getEventData(e));
+    }
+  };
+
+  render() {
+    const { value, percent } = this.state;
+    const { activeHandleID, getRailProps } = this.props;
+
+    return (
+      <Fragment>
+        {!activeHandleID && value ? (
+          <div
+            style={{
+              left: `${percent}%`,
+              position: "absolute",
+              marginLeft: "-11px",
+              marginTop: "-35px"
+            }}
+          >
+            <div className="tooltip">
+              <span className="tooltiptext">Value: {value}</span>
+            </div>
+          </div>
+        ) : null}
+        <div
+          style={railStyle}
+          {...getRailProps({
+            onMouseEnter: this.onMouseEnter,
+            onMouseLeave: this.onMouseLeave
+          })}
+        />
+        <div style={railCenterStyle} />
+      </Fragment>
+    );
+  }
+}
+
+// *******************************************************
+// HANDLE COMPONENT
+// *******************************************************
+interface HandleProps {
+  key: string;
+  handle: SliderItem;
+  isActive: Boolean;
+  disabled?: Boolean;
+  domain: number[];
+  getHandleProps: (id: string, config: object) => object;
+}
+
+export class Handle extends Component<HandleProps> {
+  static defaultProps = {
+    disabled: false
+  };
+
+  state = {
+    mouseOver: false
+  };
+
+  onMouseEnter = () => {
+    this.setState({ mouseOver: true });
+  };
+
+  onMouseLeave = () => {
+    this.setState({ mouseOver: false });
+  };
+
+  render() {
+    const {
+      domain: [min, max],
+      handle: { id, value, percent },
+      isActive,
+      disabled,
+      getHandleProps
+    } = this.props;
+    const { mouseOver } = this.state;
+
+    return (
+      <Fragment>
+        {(mouseOver || isActive) && !disabled ? (
+          <div
+            style={{
```

Comment:
can we use css classes?

Reply:
yup, added CSS classes to all the slider components :D